htdocs/contrat/list.php

```diff
@@ -774,224 +784,229 @@
 
 while ($i < min($num, $limit)) {
 	$obj = $db->fetch_object($resql);
+	if (($obj->nb_initial > 0 && in_array("1", $search_status))
```

Comment:
Adding the filtering inside the loop make the pagination system not working correctly.
Did you try to add the filter directly into the request using
$sql .= " HAVING nb_initial > 0"
or 
$sql .= " HAVING SUM(".$db->ifsql("cd.statut=0", 1, 0).') > 0"     (condition must be same than the way it is written into the select)